app/dashboard/tip_views.py

```diff
@@ -462,6 +462,15 @@ def send_tip_2(request):
             user['avatar_url'] = profile.avatar_baseavatar_related.filter(active=True).first().avatar_url
             user['preferred_payout_address'] = profile.preferred_payout_address
 
+    recent_tips = Tip.objects.filter(sender_profile=request.user.profile).order_by('-created_on')
```

Comment:
`recent_tips_profiles = Profile.objects.filter(received_tips__in=Tip.objects.filter(sender_profile__id=7766)).order_by('-received_tips__created_on')`

Unfortunately you can't use distinct here or it could be a one-liner! https://docs.djangoproject.com/en/dev/ref/models/querysets/#distinct

Anyway, no need to change the code, but thought it was an interesting way to access the profiles directly..

Reply:
Yeah, I spent 1.5+ days only on trying to make this one query better and efficient. But couldn't find a way to make it a single query/one-liner. Tried several several several ways, but no use. That's a bummer ☹️

A good one may look something like
```
recent_tips_profiles = Tip.objects.filter(sender_profile=request.user.profile).distinct('recipient_profile').order_by('-created_on')
```
But it produces this error on Postgres:
`ERROR:  SELECT DISTINCT ON expressions must match initial ORDER BY expressions`

I did find this: https://code.djangoproject.com/ticket/24218. But it's still **`new`**.

Tried **several** other ways. But had to settle with this in the end. 🤢

Reply:
yeah it really exposes where the ORM falls short, basically since you are joining tables then the rows are no longer distinct